src/Sulu/Bundle/SecurityBundle/Tests/Unit/SingleSignOn/Adapter/OpenId/OpenIdSingleSignOnAdapterTest.php

```diff
@@ -180,11 +181,30 @@ function($method, $url, $options): MockResponse {
             ->willReturn('https://sulu.io/admin');
 
         $this->userRepository->findOneBy(Argument::any())->willReturn(null);
+
+        $user = $this->prophesize(User::class);
+        $user->getRoles()->willReturn([]);
+        $user->setEmail(Argument::any())->shouldBeCalled();
+        $user->setUsername(Argument::any())->shouldBeCalled();
+        $user->setPassword(Argument::any())->shouldBeCalled();
+        $user->setSalt(Argument::any())->shouldBeCalled();
+        $user->setEnabled(Argument::any())->shouldBeCalled();
+        $user->setContact(Argument::any())->shouldBeCalled();
+        $user->addUserRole(Argument::any())->shouldBeCalled();
+        $user->getLocale()->shouldBeCalled();
+        $user->setLocale(Argument::any())->shouldBeCalled();
+
+        $contact = $this->prophesize(Contact::class);
+        $user->getContact()->willReturn($contact->reveal());
+
+        $this->userRepository->createNew()->willReturn($user->reveal());
```

Comment:
There is no need here for such complexity of mockings just return a real user entity here:

```suggestion
        $user = new User();
        $this->userRepository->createNew()->willReturn($user);
```